### PR TITLE
meta-rauc-tegra: NVIDIA Jetson TX2 (dunfell)

### PR DESCRIPTION
**Pull request by leon-anavi**
Add an example integration of RAUC with A/B slots on NVIDIA Jetson
TX2 using meta-tegra as Yocto/OE BSP layer.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

### CODE REVIEW

File: meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+        <partition name="UBENV" type="data">
+            <allocation_policy> sequential </allocation_policy>
+            <filesystem_type> basic </filesystem_type>
+            <start_location> 0x7D8000 </start_location>
+            <size> 32768 </size>
+            <file_system_attribute> 0 </file_system_attribute>
+            <allocation_attribute> 8 </allocation_attribute>
+            <percent_reserved> 0 </percent_reserved>
+            <description> **Required.** Reserved for U-Boot environment. </description>
+        </partition>
```
Comment (jluebbe) [inline meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml:175]: Does this mean that the u-boot environment is on the eMMC boot partition? Wouldn't that conflict with atomic bootloader updates?

**Comment by jluebbe** [comment]
The PR includes a `meta-rauc-tegra/recipes-core/rauc/files/.system.conf.swp`.

**Comment by leon-anavi** [comment]
Hi @jluebbe,

> The PR includes a `meta-rauc-tegra/recipes-core/rauc/files/.system.conf.swp`.

Ops, sorry. I fixed the commit.

Thank you for the feedback. Best regards, Leon

File: meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml (hunk 1 above)
Comment (leon-anavi) [reply meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml:175]: Yes, NVIDIA Jetson TX2 has [a very specific boot flow: cboot boots u-boot which boots the kernel](https://docs.nvidia.com/jetson/l4t/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/bootflow_tx2.html) and the u-boot environment is on the eMMC:
```
Tegra186 (P2771-0000-500) # saveenv
Saving Environment to MMC... Writing to MMC(0)... OK
```
I have tested A/B partitions scheme and updates of the partitions works as expected.

Best regards, Leon

Comment (jluebbe) [reply meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml:175]: Hmm, for a real product, you'd probably want to change the uboot env location to let it survive an bootloader update. For a community demo, this limitation is likely OK, but should be mentioned in a readme.

Comment (leon-anavi) [reply meta-rauc-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles/flash_l4t_t186.xml:175]: @jluebbe following your recommendation I have added a note about this in the README.rst:
https://github.com/rauc/meta-rauc-community/pull/26/commits/c815b769fb01e8ff8890db922fb36813ed983d52#diff-bc29e96693a0dca3186eeee377179357eb2ce02665f727828efea4a4b7fc4277R73

**Comment by leon-anavi** [comment]
Hi @jluebbe,

What do you think about the latest version of the patches that add a RAUC demo for NVIDIA Jetson TX2 to the master and the dunfell? Is it OK to merge them?

Thanks,
Leon

File: meta-rauc-tegra/recipes-kernel/linux/files/rauc.cfg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CONFIG_MD=y
+CONFIG_BLK_DEV_DM=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_DM_VERITY=y
+CONFIG_SQUASHFS=y
+CONFIG_CRYPTO_SHA256=y
+CONFIG_SQUASHFS_FILE_CACHE=y
```
Comment (jluebbe) [inline meta-rauc-tegra/recipes-kernel/linux/files/rauc.cfg:7]: SQUASHFS_FILE_DIRECT usually performs better. Do you have a specific reason to use _CACHE?

File: meta-rauc-tegra/recipes-kernel/linux/files/rauc.cfg [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+CONFIG_SQUASHFS_DECOMP_SINGLE=y
```
Comment (jluebbe) [inline meta-rauc-tegra/recipes-kernel/linux/files/rauc.cfg:8]: `SQUASHFS_DECOMP_MULTI` or `SQUASHFS_DECOMP_MULTI_PERCPU` usually perform better.

File: meta-rauc-tegra/recipes-core/bundles/files/development-1.cert.pem [hunk 4]
```diff
@@ -0,0 +1,80 @@
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number: 2 (0x2)
+        Signature Algorithm: sha256WithRSAEncryption
+        Issuer: O=Test Org, CN=Test Org rauc CA Development
```
Comment (jluebbe) [inline meta-rauc-tegra/recipes-core/bundles/files/development-1.cert.pem:6]: Perhaps we should have only a single copy of the key pair in the repository? But, that's for a different PR. :)

**Comment by jluebbe** [comment]
Looks generally good to me. I'd also be fine with merging as is and doing a follow-up PR. :)

**Comment by jluebbe** [comment]
@leon-anavi It looks like you lost the README comment regarding the u-boot env location in the force-push?

**Comment by leon-anavi** [comment]
I think I have addressed all discussed issues. Thank you for the feedback. I will merge the PR and we can continue with improvements through new PRs.

Thanks,
Leon

**Pull request merged**